### PR TITLE
feat: add trader-joe app

### PR DESCRIPTION
**Pull request by xdrdak**
## Checklist

- [ ] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] (optional) As a contributor, my Ethereum address/ENS is:

### CODE REVIEW

File: src/app-toolkit/app-toolkit.helpers.ts [hunk 1]
```diff
@@ -56,5 +58,7 @@ export class AppToolkitHelperRegistry {
     public readonly masterChefRewarderClaimableBalanceStrategy: MasterChefRewarderClaimableBalanceStrategy,
     @Inject(MasterChefRewarderClaimableTokenStrategy)
     public readonly masterChefRewarderClaimableTokenStrategy: MasterChefRewarderClaimableTokenStrategy,
```
Comment (immasandwich) [inline src/app-toolkit/app-toolkit.helpers.ts:60]: Can we rename this to `MasterChefV2ClaimableTokenStrategy` and `MasterChefV2ClaimableBalanceStrategy` for consistency? 🙏

File: src/app-toolkit/helpers/master-chef/master-chef.v2.reward-rate-strategy.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        const secondaryTotalRewardPerBlocks = isArray(secondaryTotalRewardPerBlock)
+          ? secondaryTotalRewardPerBlock
+          : [secondaryTotalRewardPerBlock];
+        totalRewardPerBlock.push(...secondaryTotalRewardPerBlocks);
+      }
+
+      // Calculate reward per block for the given pool's share
+      const share = Number(poolAllocPoints) / Number(totalAllocPoints);
+      const rewardPerBlock = totalRewardPerBlock.map((v, i) => share * Number(v) * Number(rewardMultiplier[i] ?? 1));
+      return rewardPerBlock;
```
Comment (immasandwich) [inline src/app-toolkit/helpers/master-chef/master-chef.v2.reward-rate-strategy.ts:83]: Anywhere you see `PerBlock` should be replace with `Rate` (considering reward rate can now be either per block or per second).

File: src/apps/trader-joe/avalanche/trader-joe.chef-boosted-farm.contract-position-fetcher.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+          multicall
+            .wrap(contract)
+            .poolInfo(poolIndex)
+            .then(v => v.rewarder),
+        resolveRewarderContract: ({ network, rewarderAddress }) =>
+          this.traderJoeContractFactory.traderJoeChefV2Rewarder({ address: rewarderAddress, network }),
+        resolveSecondaryClaimableToken: ({ multicall, rewarderContract }) =>
+          multicall.wrap(rewarderContract).rewardToken(),
+      }),
+      // resolveRewardsPerBlock
```
Comment (immasandwich) [inline src/apps/trader-joe/avalanche/trader-joe.chef-boosted-farm.contract-position-fetcher.ts:60]: Delete

File: src/apps/trader-joe/trader-joe.definition.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import { GroupType } from '~app/app.interface';
+import { ProtocolTag, ProtocolAction } from '~app/app.interface';
+import { Network } from '~types/network.interface';
+
+export const TRADER_JOE_DEFINITION = {
+  id: 'trader-joe',
+  name: 'Trader Joe',
+  description: `Trader Joe is your one-stop decentralized trading platform on the Avalanche network.`,
+  groups: {
+    pool: { id: 'pool', type: GroupType.POSITION },
```
Comment (immasandwich) [inline src/apps/trader-joe/trader-joe.definition.ts:12]: Should be `GroupType.TOKEN`

**Review by immasandwich — APPROVED** [review]
Few comments, then good to go.

File: src/apps/trader-joe/avalanche/trader-joe.balance-fetcher.ts [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+} from '../contracts';
+import { TRADER_JOE_DEFINITION } from '../trader-joe.definition';
+
+const appId = TRADER_JOE_DEFINITION.id;
+const network = Network.AVALANCHE_MAINNET;
+
+@Register.BalanceFetcher(TRADER_JOE_DEFINITION.id, Network.AVALANCHE_MAINNET)
+export class AvalancheTraderJoeBalanceFetcher implements BalanceFetcher {
+  constructor(
+    @Inject(TokenBalanceHelper) private readonly tokenBalanceHelper: TokenBalanceHelper,
```
Comment (JForsaken) [inline src/apps/trader-joe/avalanche/trader-joe.balance-fetcher.ts:32]: Most of these should be using APP_TOOLKIT for the helpers

**Pull request merged**